### PR TITLE
SNOW-1445356, SNOW-1445357: Add support for Series.str.lstrip and Series.str.rstrip

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1445356, SNOW-1445357

2. Fill out the following pre-review checklist:

   - [X] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Add support for Series.str.lstrip and Series.str.rstrip.

### CODE REVIEW

File: docs/source/modin/supported/series_str_supported.rst [hunk 1]
```diff
@@ -126,7 +126,7 @@ the method in the left column.
 | ``startswith``              | P                               |  ``N`` if the `na` parameter is set to a non-bool  |
 |                             |                                 |  value.                                            |
 +-----------------------------+---------------------------------+----------------------------------------------------+
-| ``strip``                   | Y                               |                                                    |
+| ``strip``                   | P                               | ``N`` if `to_strip` is non-string.                 |
```
Comment (sfc-gh-nkrishna) [inline docs/source/modin/supported/series_str_supported.rst:129]: Just to confirm, this isn't a regression but rather was always the case and just not documented correctly?

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, nice work!

File: src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py [hunk 2]
```diff
@@ -702,10 +702,96 @@ def strip():
...
+            Remove trailing characters in Series/Index.
+
+        Examples
+        --------
+        >>> s = pd.Series(['1. Ant.  ', '2. Bee!\\n', '3. Cat?\\t', np.nan, 10, True])
+        >>> s
+        0    1. Ant.
+        1    2. Bee!\\n
+        2    3. Cat?\\t
+        3          NaN
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py:735]:
```suggestion
        1    2. Bee!\n
        2    3. Cat?\t
        3          None
```

File: src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py [hunk 3]
```diff
@@ -702,10 +702,96 @@ def strip():
...
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py:781]:
```suggestion
        1    2. Bee!\n
        2    3. Cat?\t
        3          None
```

**Review by sfc-gh-nkrishna — COMMENTED** [review]
These suggestions should fix the docstring failures

**Review by sfc-gh-nkumar — APPROVED** [review]

File: docs/source/modin/supported/series_str_supported.rst (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply docs/source/modin/supported/series_str_supported.rst:129]: That's correct. Just improving the documentation.

**Review by devin-petersohn — APPROVED** [review]
👍

File: src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py (hunk 2 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py]: Thanks. They are fixed now.

File: src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py (hunk 3 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py]: Thanks. They are fixed now.

**Pull request merged**